### PR TITLE
Add `bumble-console --device-config` support for gatt services

### PR DESCRIPTION
**Pull request by AlanRosenthal**
This PR adds support for bumble-console to be preloaded with gatt services via `--device-config`.

This PR also adds some type annotations.

Formally #66

### CODE REVIEW

File: tests/gatt_test.py [hunk 1]
```diff
@@ -705,6 +705,57 @@ def on_d3_connection(connection):
     assert d2_connection.att_mtu == 50
 
 
+# -----------------------------------------------------------------------------
+@pytest.mark.asyncio
+async def test_char_property_to_string():
```
Comment (mogenson) [inline tests/gatt_test.py:710]: Do these tests need to be async?

File: bumble/gatt.py [hunk 2]
```diff
@@ -22,6 +22,7 @@
 # -----------------------------------------------------------------------------
 # Imports
 # -----------------------------------------------------------------------------
+from __future__ import annotations
```
[the remark this answers is not in the document]
Comment (AlanRosenthal) [reply bumble/gatt.py:25]: yeah! the issue I ran into was I wanted to use a type hint that hadn't been defined yet. see https://stackoverflow.com/a/61544901

File: tests/gatt_test.py (hunk 1 above)
Comment (AlanRosenthal) [reply tests/gatt_test.py:710]: good call out, let me redo them

File: bumble/gatt.py [hunk 3]
```diff
@@ -256,10 +257,17 @@ def properties_as_string(properties):
...
+        return functools.reduce(lambda x,y: x|get_dict_key_by_value(Characteristic.PROPERTY_NAMES, y), properties_str.split(','), 0)
+
+    def __init__(self, uuid, properties, permissions, value = b'', descriptors: list[Descriptor] = []):
         super().__init__(uuid, permissions, value)
         self.uuid        = self.type
-        self.properties  = properties
+        if type(properties) is str:
+            self.properties = Characteristic.string_to_properties(properties)
+        else:
+            self.properties  = properties
```
Comment (barbibulle) [inline bumble/gatt.py:270]: nit: extra space

File: bumble/gatt.py [hunk 4]
```diff
@@ -256,10 +257,17 @@ def properties_as_string(properties):
             if properties & p
         ])
 
-    def __init__(self, uuid, properties, permissions, value = b'', descriptors = []):
+    @staticmethod
+    def string_to_properties(properties_str: str):
+        return functools.reduce(lambda x,y: x|get_dict_key_by_value(Characteristic.PROPERTY_NAMES, y), properties_str.split(','), 0)
```
Comment (barbibulle) [inline bumble/gatt.py:262]: style: break into multiple lines and spaces after `,` and around `|`  (I suggest `flake8` to detect code style warnings, even though we're not 100% strict about that, it helps) :
```
return functools.reduce(
    lambda x, y: x | get_dict_key_by_value(Characteristic.PROPERTY_NAMES, y),
    properties_str.split(','),
    0
)
```

**Review by barbibulle — APPROVED** [review]

File: bumble/gatt.py (hunk 4 above)
Comment (AlanRosenthal) [reply bumble/gatt.py:262]: @barbibulle do you think I should verify `get_dict_key_by_value` returns non-None?

Comment (barbibulle) [reply bumble/gatt.py]: I think it's best to let this throw if you pass a property name that's not supported.

**Pull request merged**